exporter/terra/submission/exporter.py

```diff
@@ -11,37 +13,14 @@ def __init__(self, ingest_service: IngestService, terra_client: TerraTransferCli
         self.logger = logging.getLogger('TerraSubmissionExporter')
 
     def start_data_file_transfer(self, job_id: str, submission_uuid: str, project_uuid: str):
-        self.logger.info(f"data sync starting")
+        self.logger.info(f"Getting Submission for data transfer")
         submission = self.ingest_service.get_submission(submission_uuid)
-        export_data = "Export metadata" not in submission.get("submitActions", [])
-
-        self.logger.info(f"The export data flag has been set to {export_data}")
-        if export_data and not self.ingest_service.is_data_transfer_complete(job_id):
-            self.logger.info("Exporting data files..")
-            transfer_job_spec, success = self.terra_client.transfer_data_files(
-                submission,
-                project_uuid,
-                job_id
-            )
-            self._wait_for_data_transfer_to_complete(job_id, success, transfer_job_spec)
-
-    # Only the exporter process which is successful should be polling GCP Transfer service if the job is complete
-    # This is to avoid hitting the rate limit 500 requests per 100 sec https://cloud.google.com/storage-transfer/quotas
-    def _wait_for_data_transfer_to_complete(self, export_job_id, success, transfer_job_spec):
-        def compute_wait_time(start_wait_time_sec):
-            max_wait_interval_sec = 10 * 60
-            return min(start_wait_time_sec * 2, max_wait_interval_sec)
-
-        max_wait_time_sec = 60 * 60 * 6
-        start_wait_time_sec = 2
-        if success:
-            self.logger.info("Google Cloud Transfer job was successfully created..")
-            self.logger.info("Waiting for job to complete..")
-            self.terra_client.wait_for_transfer_to_complete(transfer_job_spec.name, compute_wait_time,
-                                                                  start_wait_time_sec, max_wait_time_sec)
-            self.ingest_service.set_data_transfer_complete(export_job_id)
-        else:
-            self.logger.info("Google Cloud Transfer job was already created..")
-            self.logger.info("Waiting for job to complete..")
-            self.ingest_service.wait_for_data_transfer_to_complete(export_job_id, compute_wait_time, start_wait_time_sec,
-                                                                   max_wait_time_sec)
+        if "Export" not in submission.get("submitActions", []):
+            self.logger.error("The export data flag has not been set!")
+            raise SubmissionDoesNotHaveRequiredAction()
+        self.logger.info(f"Starting data transfer")
+        upload_area = submission.get("stagingDetails", {}).get("stagingAreaLocation", {}).get("value")
+        if not upload_area:
+            self.logger.error(f"Could not find: stagingDetails.stagingAreaLocation.value: {submission}")
+            raise SubmissionDoesNotHaveStagingArea()
```

Comment:
Have you considered adding the error details to the exception?

Reply:
The log message has all of the context, because of the session context logger we are using.
It would be a useful redundancy to also have this context also stored in the exceptions, but it would still be redundant.

You can see this in the logs for this unit test:
```
2022-09-16 10:16:08,637 - TerraSubmissionExporter - INFO - submission_uuid:submissionUuid - export_job_id:exportJobId - project_uuid:projectUuid - Message received
2022-09-16 10:16:08,637 - TerraSubmissionExporter - INFO - submission_uuid:submissionUuid - export_job_id:exportJobId - project_uuid:projectUuid - Received data transfer message
2022-09-16 10:16:08,638 - TerraSubmissionExporter - INFO - submission_uuid:submissionUuid - export_job_id:exportJobId - project_uuid:projectUuid - Getting Submission for data transfer
2022-09-16 10:16:08,638 - TerraSubmissionExporter - INFO - submission_uuid:submissionUuid - export_job_id:exportJobId - project_uuid:projectUuid - Starting data transfer
2022-09-16 10:16:08,638 - TerraSubmissionExporter - ERROR - submission_uuid:submissionUuid - export_job_id:exportJobId - project_uuid:projectUuid - Could not find: stagingDetails.stagingAreaLocation.value: {'submitActions': ['Export']}
2022-09-16 10:16:08,638 - TerraSubmissionExporter - ERROR - submission_uuid:submissionUuid - export_job_id:exportJobId - project_uuid:projectUuid - Rejecting message: {"exportJobId": "exportJobId", "submissionUuid": "submissionUuid", "projectUuid": "projectUuid", "callbackLink": "callbackLink", "context": {}} due to error: SubmissionDoesNotHaveStagingArea
2022-09-16 10:16:08,639 - TerraSubmissionExporter - ERROR - submission_uuid:submissionUuid - export_job_id:exportJobId - project_uuid:projectUuid - 
Traceback (most recent call last):
  File "/Users/astaffer/Repos/ingest-exporter/exporter/queue/listener.py", line 39, in try_handle_or_reject
    self.handler.handle_message(json_body, msg)
  File "/Users/astaffer/Repos/ingest-exporter/exporter/terra/submission/handler.py", line 29, in handle_message
    self.submission_exporter.start_data_file_transfer(export.job_id, export.submission_uuid, export.project_uuid)
  File "/Users/astaffer/Repos/ingest-exporter/exporter/terra/submission/exporter.py", line 25, in start_data_file_transfer
    raise SubmissionDoesNotHaveStagingArea()
exporter.terra.exceptions.SubmissionDoesNotHaveStagingArea

```